test/e2e/infra.go

```diff
@@ -35,7 +36,34 @@ var _ = describe("Infrastructure tests", func() {
 
 	It("All node pools should be able to run pods [Zalando]", func() {
 		// When modifying this list, don't forget to modify cluster/manifests/e2e-resources/pool-reserve.yaml
-		for _, pool := range []string{"default-worker-splitaz", "worker-combined", "worker-limit-az", "worker-instance-storage"} {
+		nodePools := []string{
+			"default-worker-splitaz",
+			"worker-combined",
+			"worker-limit-az",
+			"worker-instance-storage",
+			"worker-node-tests",
+			"worker-karpenter",
+			"worker-arm64",
+		}
+
+		for _, pool := range nodePools {
+			deploy, err := cs.AppsV1().Deployments("default").Get(context.Background(), fmt.Sprintf("pool-reserve-%s", pool), metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			err = deployment.WaitForDeploymentComplete(cs, deploy)
+			framework.ExpectNoError(err)
+
+			// Scale out deployment to one more replica. In combination with Pod-Anti-Affinity, this should require one more node.
+			_, err = cs.AppsV1().Deployments("default").ApplyScale(
+				context.Background(),
+				fmt.Sprintf("pool-reserve-%s", pool),
+				applyconfigurationsautoscalingv1.Scale().WithSpec(applyconfigurationsautoscalingv1.ScaleSpec().WithReplicas(2)),
+				metav1.ApplyOptions{FieldManager: "e2e.test", Force: true},
+			)
```

Comment:
Does this correctly wait for the pod to be scheduled to verify it got a node? :thinking:

Reply:
Sort of. It applies the seven Deployments sequentially and then waits for all seven in the next loop. By the time it enters the next loop, the first deployment in the list already has a pending pod.

But I can also add a small time delay to be sure. (Would skip running the tests though to speed up the merge)

Reply:
To clarify: This call doesn't wait. The wait happens in the next loop below.

Reply:
I guess I was/am unsure if `err = deployment.WaitForDeploymentComplete(cs, deploy)` waits for deployment to have 2 ready pods.

Reply:
Yes, I can confirm that it does that.